### PR TITLE
Modernize project setup

### PR DESCRIPTION
**Pull request by dlecocq**
Update this to have:

- vagrant
- setup.cfg
- meaningful requirements.txt

### CODE REVIEW

**Comment by dlecocq** [comment]
@neilmb @b4hand @lindseyreno

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# http://docs.vagrantup.com/v2/
+Vagrant.configure('2') do |config|
+  config.vm.box = 'ubuntu/trusty64'
+  config.vm.hostname = 'url-py'
+  config.ssh.forward_agent = true
+
+  config.vm.provider :virtualbox do |vb|
+    vb.customize ["modifyvm", :id, "--memory", "1024"]
+    vb.customize ["modifyvm", :id, "--cpus", "2"]
```
Comment (vadim-moz) [inline Vagrantfile]: @miroswan [suggested adding](https://seomoz.atlassian.net/browse/ROGER-1158):
```rb
vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
```
to ensure correct work of VPN. I did not have a chance to test it, but I used to have problems running VPN from home.

**Comment by lindseyreno** [comment]
LGTM

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,13 @@
-clean:
-	# Remove the build
-	rm -rf url.egg-info build dist
-	# And all of our pyc files
-	find . -name '*.pyc' | xargs -n 100 rm
-	# And lastly, .coverage files
-	find . -name .coverage | xargs rm
+.PHONY: test
+test:
```
Comment (vadim-moz) [inline Makefile]: lint / pep8?

Personally, I would prefer `lint` as a separate target.

File: Makefile [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+	nosetests --with-coverage
```
Comment (vadim-moz) [inline Makefile]: may I recommend `--with-tmer` and `--rednose`? They seems worthy to me.

File: scripts/vagrant/provision.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    pip install virtualenv
+    if [ -d venv ]; then
+        virtualenv venv
+    fi
+    source venv/bin/activate
+
+    # Lastly, our dependencies
+    pip install -r requirements.txt
+
+popd
```
Comment (b4hand) [inline scripts/vagrant/provision.sh]: Missing trailing newline.

File: .gitignore [hunk 5]
```diff
@@ -2,4 +2,6 @@
 build/
 dist/
 url.egg-info/
-.coverage
\ No newline at end of file
+.coverage
+venv
+.vagrant
```
Comment (b4hand) [inline .gitignore]: Missing trailing newline.

File: Vagrantfile (hunk 1 above)
Comment (dlecocq) [inline Vagrantfile]: Nodes

File: Makefile (hunk 2 above)
Comment (dlecocq) [inline Makefile]: I'm not sure how linting works with Cython -- perhaps we can revisit this after the cython switch.

File: Makefile (hunk 3 above)
Comment (dlecocq) [inline Makefile]: We've got `rednose` and other options turned on in `setup.cfg`, but `--with-timer` is a good idea.

File: scripts/vagrant/provision.sh (hunk 4 above)
Comment (dlecocq) [inline scripts/vagrant/provision.sh]: Noted

File: .gitignore (hunk 5 above)
Comment (dlecocq) [inline .gitignore]: Noted

**Comment by dlecocq** [comment]
Reupped

**Pull request merged**